bin/boilerplate/_extras/figures.md

```diff
@@ -5,22 +5,55 @@ title: Figures
   window.onload = function() {
     var lesson_episodes = [
     {% for episode in site.episodes %}
-    "{{ episode.url}}"{% unless forloop.last %},{% endunless %}
+    "{{ episode.url }}"{% unless forloop.last %},{% endunless %}
     {% endfor %}
     ];
+
     var xmlHttp = [];  /* Required since we are going to query every episode. */
     for (i=0; i < lesson_episodes.length; i++) {
+
       xmlHttp[i] = new XMLHttpRequest();
       xmlHttp[i].episode = lesson_episodes[i];  /* To enable use this later. */
       xmlHttp[i].onreadystatechange = function() {
+
         if (this.readyState == 4 && this.status == 200) {
-          var article_here = document.getElementById(this.episode);
           var parser = new DOMParser();
           var htmlDoc = parser.parseFromString(this.responseText,"text/html");
           var htmlDocArticle = htmlDoc.getElementsByTagName("article")[0];
-          article_here.appendChild(htmlDocArticle.getElementsByTagName("h1")[0]);
-          for (let image of htmlDocArticle.getElementsByTagName("img")) {
-            article_here.appendChild(image);
+
+          var article_here = document.getElementById(this.episode);
+          var images = htmlDocArticle.getElementsByTagName("img");
+
+          if (images.length > 0) {
+            var h1text = htmlDocArticle.getElementsByTagName("h1")[0].innerHTML;
+
+            var htitle = document.createElement('h2');
+            htitle.innerHTML = 'Episode "' + h1text + '"';
+            htitle.style = "margin-top: 100px;";
```

Comment:
Can we use a class instead of inline CSS?

Reply:
I don't know of a dedicated class for H2, H3, etc. in SWC/DC lessons (only for H1). I can create one, though, like `.figure-title`.

Reply:
In `_extras/figures.md` you have

```
{% for episode in site.episodes %}
<article id="{{ episode.url }}"></article>
{% endfor %}
```

you can change it to

```
{% for episode in site.episodes %}
<article id="{{ episode.url }} class="figures"></article>
{% endfor %}
```

and add a rule to the CSS file like

```
article.figures h1,
article.figures h2,
article.figures h3,
article.figures h4,
article.figures h5,
article.figures h6 {
    ...
}
```

I think this approach is better because contributors to the lesson should have some knowledge of HTML to be able to read the file and get one idea of what it is doing but we can't expect that contributors will be confident in read Javascript.